python/dgl/graphbolt/subgraph_sampler.py

```diff
@@ -226,6 +233,94 @@ def sampling_stages(self, datapipe):
         """
         return datapipe.transform(self._sample)
 
+    @staticmethod
+    def _seeds_preprocess(minibatch):
+        use_timestamp = hasattr(minibatch, "timestamp")
+        seeds = minibatch.seeds
+        is_heterogeneous = isinstance(seeds, Dict)
+        if is_heterogeneous:
+            # Collect nodes from all types of input.
+            nodes = defaultdict(list)
+            nodes_timestamp = None
+            if use_timestamp:
+                nodes_timestamp = defaultdict(list)
+            for etype, pair in seeds.items():
```

Comment:
Ideally the code structure should be:
```
def handle_homo ...

if is_hetero:
  results = {}
  for each type:
    results[type] = handle_homo()
else
  result = handle_homo()
```
Is it possible to refactor the code here?